services/agents/vmagent.go

```diff
@@ -32,6 +32,8 @@ func vmAgentConfig(scrapeCfg string) *agentpb.SetStateRequest_AgentProcess {
 		"-promscrape.config={{.TextFiles.vmagentscrapecfg}}",
 		// 1GB disk queue size.
 		"-remoteWrite.maxDiskUsagePerURL=1073741824",
+		// 1 hour timeout for sending data.
+		"-remoteWrite.sendTimeout=1h",
```

Comment:
Did you check how much memory it can consume? And what happens if it conflicts with `remoteWrite.maxDiskUsagePerURL` above?